airbyte-db/db-lib/src/main/java/io/airbyte/db/jdbc/DefaultJdbcDatabase.java

```diff
@@ -73,10 +74,12 @@ public <T> Stream<T> unsafeResultSetQuery(final CheckedFunction<Connection, Resu
 
   @Override
   public DatabaseMetaData getMetaData() throws SQLException {
-    final Connection conn = dataSource.getConnection();
-    final DatabaseMetaData metaData = conn.getMetaData();
-    conn.close();
-    return metaData;
+    try (final Connection connection = dataSource.getConnection()) {
+      final DatabaseMetaData metaData = connection.getMetaData();
+      return metaData;
+    } catch (SQLException e) {
+      throw new ConnectionErrorException(e.getSQLState(), e.getCause() == null ? e : e.getCause());
+    }
```

Comment:
How come this method no longer closes out the connection? It seems important to have a `finally` block that closes out the connection to make sure connections are not left idling, right?

Reply:
> How come this method no longer closes out the connection? It seems important to have a `finally` block that closes out the connection to make sure connections are not left idling, right?

please pay attention here we use _try-with-resources_ statement which ensures that each resource is closed at the end of the statement

Reply:
Thanks for the catch, my mind mistakenly read that as 
```java
try {
  final Connection conn = dataSource.getConnection()
  ...
} catch (SQLException e) {}
```